DoorGate/ViewController.swift

```diff
@@ -24,34 +26,33 @@ class ViewController: UIViewController {
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        
+
         //In Button Logic
-        inButton.rx.tap.scan(0) { (priorValue, _) in
-                return 1
-            }
-            .subscribe(onNext: { [unowned self] currentCount in
-                let count = Int(self.countLabel.text!)! + currentCount
-                self.logic(count: count)
-                })
+        inButton.rx.tap
+            .subscribe({ [unowned self] currentCount in
+                self.presenter.buttonTap(command: DoorCommand.goingIn)
+            })
             .disposed(by: disposeBag)
-       
         //Out Button Logic
-        outButton.rx.tap.scan(0) { (priorValue, _) in
-                return -1
-            }
-            .subscribe(onNext: { [unowned self] currentCount in
-                let count = Int(self.countLabel.text!)! + currentCount
-                self.logic(count: count)
+        outButton.rx.tap
+            .subscribe({ [unowned self] currentCount in
+                self.presenter.buttonTap(command: DoorCommand.goingOut)
+                })
+            .disposed(by: disposeBag)
+
+        presenter.doorEntity.state.asObservable()
+            .subscribe(onNext: { state in
+                //This feels wrong empasulating self without the [unowned self]
+                self.updateUI(count: state)
             })
             .disposed(by: disposeBag)
     }
-    
-    func logic(count :Int) {
+
+    func updateUI(count :Int) {
         self.countLabel.text = String(count)
         self.stateLabel.text = stateString[count]
-
+        
         self.inButton.isEnabled = (count < 2)
```

Comment:
might create constants here equal to 2 and 0 and name them soemthing like THRESHOLD_TO_DISABLE_IN and THRESHOLD_TO_DISABLE_OUT  or something to that effect to make the purpose of the code more implicit.

Reply:
yea, definitely. It would be easier to read